pkg/dogstatsd/server.go

```diff
@@ -711,21 +710,17 @@ func (s *Server) storeMetricStats(sample metrics.MetricSample) {
 	s.Debug.Lock()
 	defer s.Debug.Unlock()
 
-	if s.debugTagsAccumulator == nil {
-		s.debugTagsAccumulator = oldtagset.NewHashingTagsAccumulator()
-	}
+	tags := tagset.NewTags(sample.Tags)
 
 	// key
-	defer s.debugTagsAccumulator.Reset()
-	s.debugTagsAccumulator.Append(sample.Tags...)
-	key := s.Debug.keyGen.Generate(sample.Name, "", s.debugTagsAccumulator)
+	key := s.Debug.keyGen.Generate(sample.Name, "", tags)
 
 	// store
 	ms := s.Debug.Stats[key]
 	ms.Count++
 	ms.LastSeen = now
 	ms.Name = sample.Name
-	ms.Tags = strings.Join(s.debugTagsAccumulator.Get(), " ") // we don't want/need to share the underlying array
+	ms.Tags = tags.String()
```

Comment:
Note: Before  the separator was `" "` whereas now it is `", "`. As it is a debug function I don't think that is an issue.

Reply:
Oh, nicely spotted.  I agree that this is probably not an issue, now that you've noticed.

Reply:
..on the other hand, it's tested -- I'll fix it.